app/mapping/ClassOfBeneficiariesMapper.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mapping
+
+import models.UserAnswers
+import viewmodels.addAnother.ClassOfBeneficiaryViewModel
+import viewmodels.ClassOfBeneficiaries
+
+
+class ClassOfBeneficiariesMapper extends Mapping[List[UnidentifiedType]]{
+
+  override def build(userAnswers: UserAnswers): Option[List[UnidentifiedType]] = {
+    val beneficiaries : List[ClassOfBeneficiaryViewModel] = userAnswers.get(ClassOfBeneficiaries).getOrElse(List.empty[ClassOfBeneficiaryViewModel])
+    beneficiaries match {
+      case Nil => None
+      case list => Some(list.map {
+        case beneficiary : ClassOfBeneficiaryViewModel =>
+          UnidentifiedType(
+            description = beneficiary.description.get,
+            beneficiaryDiscretion = None,
+            beneficiaryShareOfIncome = None
```

Comment:
@jassalrichy do we need to map over discretion and share on income?

Reply:
Resolved

Reply:
No